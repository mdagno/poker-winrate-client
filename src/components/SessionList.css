.sessionList{
  color: white;
  display: flex;
  flex-direction: column;
  margin: 10px;
  margin-left: auto;
  margin-right: auto;
  width: 80%;
  text-align: center;

}

.sessionList ul{
  list-style: none;
  padding: 0;
}

.sessionList li {
  border: 2px solid white;
  border-radius: 2px;
  margin: 5px;
}

.sessionList li:hover{
  background-color: white;
}

.sessionList select{
  width: 60%;
  margin-left: auto;
  margin-right: auto;
}

.sessionList a {
  text-decoration: none;
  color:  #09d3ac;
}