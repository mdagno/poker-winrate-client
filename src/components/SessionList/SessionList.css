.sessionList{
  color: white;
  display: flex;
  flex-direction: column;
  margin: 10px;
  margin-left: auto;
  margin-right: auto;
  width: 80%;
  text-align: center;
  margin-top: 5%;
}

.sessionList ul{
  list-style: none;
  padding: 0;
  font-size: 1em;
}

.sessionList li h3{
  color: #222;
  font-size: 1em;
  margin: 0.3em;
}

.sessionList li p {
  font-size: 1em;
}

.sessionList li {
  border-radius: 10px;
  border: 2px outset white;
  margin: 8px;
  background-color: #ccc;
  padding: 0.5em;
  box-shadow: inset -3px -3px 3px 0px #475080;
}

.sessionList select{
  width: 60%;
  margin-left: auto;
  margin-right: auto;
  display: block;
	font-size: 16px;
	font-family: sans-serif;
	font-weight: 700;
	color: #222;
	line-height: 1.3;
	padding: 0.5em;
	border: 1px solid #aaa;
	box-shadow: 0 1px 0 1px rgba(0,0,0,.04);
	border-radius: .5em;
	-moz-appearance: none;
	-webkit-appearance: none;
	appearance: none;
	background-color: #fff;
	background-image: url('data:image/svg+xml;charset=US-ASCII,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%22292.4%22%20height%3D%22292.4%22%3E%3Cpath%20fill%3D%22%23007CB2%22%20d%3D%22M287%2069.4a17.6%2017.6%200%200%200-13-5.4H18.4c-5%200-9.3%201.8-12.9%205.4A17.6%2017.6%200%200%200%200%2082.2c0%205%201.8%209.3%205.4%2012.9l128%20127.9c3.6%203.6%207.8%205.4%2012.8%205.4s9.2-1.8%2012.8-5.4L287%2095c3.5-3.5%205.4-7.8%205.4-12.8%200-5-1.9-9.2-5.5-12.8z%22%2F%3E%3C%2Fsvg%3E'),
	  linear-gradient(to bottom, #ffffff 0%,#e5e5e5 100%);
	background-repeat: no-repeat, repeat;
	background-position: right .7em top 50%, 0 0;
	background-size: .65em auto, 100%;
}

.sessionList a {
  text-decoration: none;
  color: #222;
}

.sessionList .sessionInfo {
  display: flex;
  flex-direction: row;
  border-radius: 10px;
}

.sessionInfo li {
  margin: 0;
  border-radius: 2px;
  border: 2px inset white;
  width: 33%;
  font-size: 1em;
  box-shadow: none;
}

.sessionInfo p {
  margin: 0;
}

.sessionInfo .winning {
  background-color: #009BB0;
  color: white;
  font-style: bold;
  font-size: 1em;
  padding-top: 0.8em;
}

.sessionInfo .losing {
  background-color: #B31731;
  color: white;
  font-style: bold;
  font-size: 1em;
  padding-top: 0.8em;
}

#tournament {
  color: white;
  background-color: black;
  font-size: 0.8em;
  padding-top: 1.2em;
}

#buyIn {
  color: black;
  background-color: white;
  font-size: 0.8em;
}

#bigBlind {
  padding-top: 0.7em;
}

  .sessionList li:hover{
    background-color: #00866F;
  }

@media only screen and (min-device-width : 768px) {
  .sessionList{
    width: 70%;
  }

  .sessionList-list {
    padding-top: 1em;
  }
  
  .sessionInfo .winning {
    padding: 0.8em;
    font-size: 1.2em;
  }
  
  .sessionInfo .losing {
    padding: 0.8em;
    font-size: 1.2em;
  }
  
  #buyIn{
    font-size: 1.08em;
  }

  #tournament {
    font-size: 1em;
    text-align: center;
  }
}

/* Desktops and laptops ----------- */
@media only screen  and (min-width : 1024px) {
  .sessionList{
    display: flex;
    width: 100%;
    margin-top: 80px;
  }

  .sessionList li{
    margin-bottom: 10px;
    width: 32%;
    display: inline-block;
  }


  .sessionList{
    width: 60%;
  }

  .sessionList li h3{
   font-size: 1.25em;
  }

  .sessionInfo .winning {
    padding: 0.8em;
    font-size: 1.2em;
  }
  
  .sessionInfo .losing {
    padding: 0.8em;
    font-size: 1.2em;
  }
  
  #buyIn{
    padding-top: 1em;
    font-size: 1.08em;
  }

  #bigBlind {
    padding-top: 0.9em;
    font-size: 1.08em;
  }

  #tournament {
    font-size: 1em;
    text-align: center;
  }
}

@media only screen and (min-device-width : 768px) and (max-device-width: 1024px) {
  .sessionList {
    margin-top: 10%;
  }
  
}
