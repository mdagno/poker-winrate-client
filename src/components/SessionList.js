import React from 'react';
import './SessionList.css';
import ApiContext from '../ApiContext'
import ApiService from '../services/api-service'
import { Link } from 'react-router-dom';

export default class SessionList extends React.Component {
  static contextType = ApiContext;

  renderSessions = () => {
    const {sessionsList = [] } = this.context;
    return sessionsList.map((session, index) => {
      return (
        <li key={index}>
          <Link to={`/sessions/${session.id}`}>
          <h3>{session.date_played}</h3>
          </Link>
          <p>Big Blind: ${session.big_blind}</p>
          <p>Buy-in: ${session.buy_in}</p>
          <p>Cashed out: ${session.cashed_out}</p>
        </li>
      )
    })
  }

  render(){
    return(
      <div className='sessionList'>
      <h2>Sessions</h2>
      <select>
        <option value=''>Filter</option>
      </select>
      <ul className='sessionList-list'>
        {this.renderSessions()}
      </ul>
      </div>
    )
  }
}