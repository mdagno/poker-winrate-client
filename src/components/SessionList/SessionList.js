import React from 'react';
import './SessionList.css';
import ApiContext from '../../ApiContext'
import ApiService from '../../services/api-service'
import { Link } from 'react-router-dom';
import moment from 'moment';

export default class SessionList extends React.Component {
  state = {
    sortBy: 'Date'
  }
  static contextType = ApiContext;

  handleFilterChange = value => {
    this.setState({
      sortBy: value
    })
    console.log('Filter executing...')
  }

  componentDidMount() {
    ApiService.getSessions()
    .then(this.context.setSessionsList)
  }

  renderSessions = () => {
    const {sessionsList = [] } = this.context;
    let renderedSessions = sessionsList;
    let sortBy = this.state.sortBy;
    
    if(sortBy === 'Online') {
      renderedSessions = renderedSessions.filter(session => session.game_type_one === 'Online')
    }
    else if(sortBy === 'Live') {
      renderedSessions = renderedSessions.filter(session => session.game_type_one === 'Live')
    }
    else if(sortBy === 'Stake'){
      renderedSessions = renderedSessions.sort((a,b) => b.big_blind - a.big_blind)
    }
    else if(sortBy === 'Date'){
      renderedSessions = renderedSessions.sort((a,b) =>  b.date_played - a.date_played)
    }
    else if(sortBy === 'Time Played'){
      renderedSessions = renderedSessions.sort((a,b) => b.session_length - a.session_length)
    }

    return renderedSessions.map((session, index) => {
      return (
        <li key={index}>
          <Link to={`/sessions/${session.id}`}>
          <h3>{moment(session.date_played).format('MMMM Do YYYY, h:mm:ss a')}</h3>
          <ul className='sessionInfo'>
            <li>BB: ${session.big_blind}</li>
            {
              (session.cashed_out > session.buy_in) ? <li className='winning'>+${session.cashed_out-session.buy_in}</li> : <li className='losing'>-${Math.abs(session.cashed_out-session.buy_in)}</li>
            }
            <li>Buy-in: ${session.buy_in}</li>
          </ul>
          </Link>
        </li>
      )
    })
  }

  render(){
    console.log(this.state.sortBy)
    return(
      <div className='sessionList'>
      <h2>Your Sessions</h2>
      <label htmlFor='filterSessions'>Filter By: </label>
      <select onChange={e => this.handleFilterChange(e.target.value)}>
        <option value='Date'>Most Recent</option>
        <option value='Live'>Live</option>
        <option value='Online'>Online</option>
        <option value='Stake'>Stake</option>
        <option value='Time played'>Time played</option>
      </select>
      <ul className='sessionList-list'>
        {this.renderSessions()}
      </ul>
      </div>
    )
  }
}