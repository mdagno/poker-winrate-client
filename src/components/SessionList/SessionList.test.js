import React from 'react';
import SessionList from './SessionList';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';

describe('SessionList component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SessionList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('SessionList renders correctly', () => {
    const tree = renderer
      .create(<SessionList />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})


