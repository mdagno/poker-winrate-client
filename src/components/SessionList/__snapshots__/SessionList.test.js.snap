// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionList component SessionList renders correctly 1`] = `
<div
  className="sessionList"
>
  <h2>
    Your Sessions
  </h2>
  <label
    htmlFor="filterSessions"
  >
    Filter By: 
  </label>
  <select
    onChange={[Function]}
  >
    <option
      value="Date"
    >
      Most Recent
    </option>
    <option
      value="Live"
    >
      Live
    </option>
    <option
      value="Online"
    >
      Online
    </option>
    <option
      value="Stake"
    >
      Stake
    </option>
    <option
      value="Time played"
    >
      Time played
    </option>
  </select>
  <ul
    className="sessionList-list"
  />
</div>
`;
